### PR TITLE
Simplify CacheKey used for scripts

### PR DESCRIPTION
**Pull request by jdconrad**
Replaced the CacheKey class with a static method that returns a String.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/script/ScriptService.java [hunk 1]
```diff
@@ -598,32 +598,9 @@ public String toString() {
...
-            return lang.equals(other.lang) && script.equals(other.script);
-        }
-
-        @Override
-        public int hashCode() {
-            return lang.hashCode() + 31 * script.hashCode();
-        }
+    private static String getCacheKey(ScriptEngineService scriptEngineService, String script) {
+        String lang = scriptEngineService.types()[0];
+        return lang + script;
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/script/ScriptService.java]: Maybe put a marker between them, like '$' or ':'?

**Comment by rjernst** [comment]
LGTM. Do we have any tests for the cache?

File: core/src/main/java/org/elasticsearch/script/ScriptService.java (hunk 1 above)
Comment (jdconrad) [inline core/src/main/java/org/elasticsearch/script/ScriptService.java]: Done.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review.  Added a test specifically for this, but previously created tests also offer significant coverage.

**Pull request closed without merging**